src/EditorFeatures/Core/Shared/Extensions/ITextBufferEditExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
+{
+    internal static class ITextBufferEditExtensions
+    {
+        private static Exception s_lastException = null;
+
+        /// <summary>
+        /// Logs exceptions thrown during <see cref="ITextBufferEdit.Apply"/> as we look for issues.
+        /// </summary>
+        /// <param name="edit"></param>
+        /// <returns></returns>
+        public static ITextSnapshot ApplyAndLogExceptions(this ITextBufferEdit edit)
+        {
+            try
+            {
+                return edit.Apply();
+            }
+            catch (Exception e) when (ErrorReporting.FatalError.ReportWithoutCrash(e))
```

Comment:
So no, the editor isn't logging but we're going to get a bug on them to do that. It occurred to me I don't have confidence that we don't swallow exceptions in a few places ourselves that might swallow the original corruption and prevent it from getting to the editor. I guess I'd rather over-telemetry in this case than under-telemetry. Once we have an answer for the root bug I'll revert this either way.

@Pilchie @jmarolf Could you give a review and approve/disapprove as you see fit?